pkg/services/authn/authnimpl/priority_queue.go

```diff
@@ -0,0 +1,33 @@
+package authnimpl
+
+type priority interface {
+	Priority() uint
+}
+
+func newQueue[T priority]() *queue[T] {
+	return &queue[T]{items: []T{}}
+}
+
+type queue[T priority] struct {
+	items []T
+}
+
+func (q *queue[T]) insert(c T) {
```

Comment:
```suggestion
func (q *queue[T]) insert(c T, priority int) {
```

maybe we can use it with `any` and without the priority method (defining the priority when we register them) so we can use this for post auth hooks as well?